packages/react-performance/README.md

```diff
@@ -146,7 +277,80 @@ function ProductPage() {
 
 #### usePerformanceReport
 
-> Todo
+A custom hook which takes a url and sends a detailed breakdown of all navigations and events which took place while the page was loading. This hook should generally only be used once at the top of your component tree.
+
+```tsx
+// App.tsx
+import React from 'react';
+import {usePerformanceReport} from '@shopify/react-performance';
+import {ProductPage} from './ProductPage';
+
+function App() {
+  usePerformanceReport('/performance-report');
+  return <ProductPage />;
+}
+```
+
+To get sensical data, applications using `<PerformanceReport />` should be sure to have at least on performance mark on each top level page.
+
+```tsx
+// ProductPage
+import React from 'react';
+import {usePerformanceMark} from '@shopify/react-performance';
+
+function ProductPage() {
+  usePerformanceMark('complete');
+
+  return <div>cool product page</div>;
+}
+```
+
+The data sent to the `/performance-report` url should look something like:
```

Comment:
This is an implementation detail, since we offer abstractions on both end we should not actively show the implementation details between them